tests/test_auth.py

```diff
@@ -44,3 +45,36 @@ def test_no_auth(self):
             assert j['url'] == url
             assert 'user' not in r.text
             assert 'passwd' not in r.text
+
+
+class TestGuessProxyAuth(unittest.TestCase):
+
+    @mock.patch('requests_toolbelt.auth.http_proxy_digest.HTTPProxyDigestAuth.handle_407')
+    def test_handle_407_header_digest(self, mock_handle_407):
+        r = requests.Response()
+        r.headers['Proxy-Authenticate'] = 'Digest nonce="d2b19757d3d656a283c99762cbd1097b", opaque="1c311ad1cc6e6183b83bc75f95a57893", realm="me@kennethreitz.com", qop=auth'
+
+        guess_auth = GuessProxyAuth(None,None,"user","passwd")
+        guess_auth.handle_407(r)
+
+        mock_handle_407.assert_called_with(r)
+
+
+    @mock.patch('requests.auth.HTTPProxyAuth.__call__')
+    @mock.patch('requests.cookies.extract_cookies_to_jar')
+    def test_handle_407_header_basic(self, mock_cookiejar, mock_proxyAuth_call):
```

Comment:
Can we just rename these to `extract_cookies_to_jar` and `proxy_auth_call` instead of `mock_*`?